skorch/tests/test_net.py

```diff
@@ -1444,3 +1444,14 @@ def get_step(self):
         expected_losses = list(
             flatten(net.history[:, 'batches', :, 'train_loss']))
         assert np.allclose(side_effect[2::3], expected_losses)
+
+    def test_train_valid_dataset(self, net_cls, module_cls, data, dataset_cls):
```

Comment:
What is missing somewhat from these tests is to show that `train_ds` is actually used for training and `valid_ds` for validation. This could be achieved, e.g., by mocking `get_iterator` and tracking what it is called with.